provider/cmd/crd2pulumi/gen/dotnet.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gen
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/pkg/v2/codegen"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/dotnet"
+	pschema "github.com/pulumi/pulumi/pkg/v2/codegen/schema"
+)
+
+var unneededDotNetFiles = []string{
+	"Provider.cs",
+	"Meta/V1/Inputs/ObjectMetaArgs.cs",
+	"Meta/V1/Outputs/ObjectMeta.cs",
+	"Meta/V1/README.md",
+	"Meta/README.md",
+}
+
+func (pg *PackageGenerator) genDotNet(types map[string]pschema.ObjectTypeSpec, baseRefs []string) (map[string]*bytes.Buffer, error) {
+	// For some reason, the Metadata input type would be generated as
+	// `ObjectMeta` instead of `ObjectMetaArgs` if we did not have an actual
+	// ObjectMeta type defined. So we'll create a placeholder metadata spec
+	// in the Pulumi schema, and then manually remove the generated files for
+	// the type later.
+	AddPlaceholderMetadataSpec(types)
+
+	pkg, err := getPackage(types, baseRefs)
+	if err != nil {
+		fmt.Print(errors.Wrap(err, "could not create package"))
+	}
+
+	// Set up C# namespaces
+	namespaces := map[string]string{}
+	for _, groupVersion := range pg.GroupVersions {
+		group, version := splitGroupVersion(groupVersion)
+		groupPrefix := groupPrefix(group)
+		namespaces[groupVersion] = strings.Title(groupPrefix) + "." + versionToUpper(version)
+	}
+	namespaces["meta/v1"] = "Meta.V1"
+
+	// Configure C# language-specific settings. Notice that we set
+	// `compatibility` to `kubernetes20`. This is because the actual ObjectMeta
+	// class defined in the .NET SDK is located at
+	// `Pulumi.Kubernetes.Types.Outputs.Meta.V1.ObjectMeta`. This path would
+	// only get generated properly if `compatibility` was `kubernetes20`.
+	pkg.Language["csharp"] = rawMessage(map[string]interface{}{
+		"compatibility":          "kubernetes20",
+		"dictionaryConstructors": true,
+		"namespaces":             namespaces,
+	})
+
+	files, err := dotnet.GeneratePackage(tool, pkg, nil)
```

Comment:
We probably also want to customize the name of the resulting `Pulumi.Kubernetes.csproj` file that is emitted. Ultimately, for https://github.com/pulumi/pulumi-kubernetes/issues/1261, we're going to want to have a package name based on each community operator name. We need to come up with a naming scheme for these. For .NET, maybe these will be something like `Pulumi.Kubernetes.Crds.<operator>` (and similar for other languages like `@pulumi/kubernetes-crds-<operator>` for Node.js, `pulumi_kubernetes_crds_<operator>` for Python, etc.).

I see for Python you're using `pulumi_crds`. Perhaps for now this file should be `Pulumi.Crds.csproj`?

Reply:
I set the package name to be Kubernetes across all languages, so it should be `pulumi_kubernetes` for Python now.